packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1290,10 +1322,6 @@ export class MergeTree {
 			pendingSegmentGroup.segments.map((pendingSegment) => {
 				const overlappingRemove = !pendingSegment.ack(pendingSegmentGroup, opArgs);
 				overwrite = overlappingRemove || overwrite;
-
-				if (!overlappingRemove && opArgs.op.type === MergeTreeDeltaType.REMOVE) {
-					this.slideAckedRemovedSegmentReferences(pendingSegment);
```

Comment:
this still need to happen as part of ack, just later (so that they slide to their eventual destination immediately)